Modules/Typescript/Typescript-Features/homework.js

```diff
@@ -0,0 +1,113 @@
+console.log(`
+  Task
+    1. Create class, which extends from an AbstractClass
+    2. Apply interfaces to class, it's method, property
+    3. Use generics there
+    4. Analyze transpiled code in ES5 and explain it
+`);
+var LimbEnum;
+(function (LimbEnum) {
+    LimbEnum["WING"] = "Wing";
+    LimbEnum["PAW"] = "Paw";
+    LimbEnum["TAIL"] = "Tail";
+    LimbEnum["HORN"] = "Horn";
+})(LimbEnum || (LimbEnum = {}));
+var LimbPossitionEnum;
+(function (LimbPossitionEnum) {
+    LimbPossitionEnum["BACK"] = "Back";
+    LimbPossitionEnum["LEFT_TOP"] = "Left Top";
+    LimbPossitionEnum["LEFT_BOTTOM"] = "Left Bottom";
+    LimbPossitionEnum["RIGHT_TOP"] = "Right Top";
+    LimbPossitionEnum["RIGHT_BOTTOM"] = "Right Bottom";
+    LimbPossitionEnum["FOREHEAD"] = "Forehead";
+})(LimbPossitionEnum || (LimbPossitionEnum = {}));
+class Animal {
```

Comment:
there are no `interfaces` in javascript

Reply:
ES5 didn't have classes, you need to change your target to generate ES5 code

Reply:
(y) changed to `ES5`